src/sentry/tagstore/snuba/backend.py

```diff
@@ -205,25 +202,32 @@ def __get_tag_key_and_top_values(
             tenant_ids=tenant_ids,
         )
 
-        if raise_on_empty and (not result or totals.get("count", 0) == 0):
-            raise TagKeyNotFound if group is None else GroupTagKeyNotFound
+        if raise_on_empty:
+            if not result or totals.get("count", 0) == 0:
+                raise TagKeyNotFound if group is None else GroupTagKeyNotFound
+
+            # Treat keys that only have empty-string values as "not found" for the
+            # get_tag_key/get_group_tag_key family of calls.
+            has_non_empty_value = any(value != "" for value in result.keys())
+            if not has_non_empty_value:
+                raise TagKeyNotFound if group is None else GroupTagKeyNotFound
```

Comment:
this is the only real change that was breaking some tests. I think they're requesting specific tags and we want to throw if they're not found?